MoreLinq/Values.cs

```diff
@@ -0,0 +1,63 @@
+#region License and Terms
+// MoreLINQ - Extensions to LINQ to Objects
+// Copyright (c) 2015 Felipe Sateler. All rights reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#endregion
+
+namespace MoreLinq
+{
+    using System;
+    using System.Collections.Generic;
+    using LinqEnumerable = System.Linq.Enumerable;
+
+    public static partial class MoreEnumerable
+    {
+        /// <summary>
+        /// Invokes a sequence of factory functions in order, and produces a sequence of the results
+        /// </summary>
+        /// <remarks>
+        /// This operator uses deferred execution and streams the results.
+        /// If the resulting sequence is enumerated multiple times, the factories will be
+        /// invoked multiple times too
+        /// </remarks>
+        /// <typeparam name="T">The type of the object returned by the factories</typeparam>
+        /// <param name="factories">The factories to evaluate</param>
+        /// <returns>A sequence with the values created for all of the <paramref name="factories"/></returns>
+        /// <exception cref="ArgumentNullException">When <paramref name="factories"/> is <c>null</c></exception>
+        public static IEnumerable<T> Values<T>(params Func<T>[] factories)
```

Comment:
This would mean the `params` version would have a different name from the extension method, right? I don't think `var values = funcs.From();` makes much sense.